src/app/components/SubjectHeading/SubjectHeadingSearch.jsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import { Link } from 'react-router';
+import PropTypes from 'prop-types';
+import axios from 'axios';
+import appConfig from '../../../../appConfig';
+import AutosuggestItem from './AutosuggestItem'
+
+class SubjectHeadingSearch extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      activeSuggestion: 0,
+      suggestions: [],
+      userInput: ''
+    }
+
+    this.onChange = this.onChange.bind(this)
+    this.onSubmit = this.onSubmit.bind(this)
+    this.clearUserInput = this.clearUserInput.bind(this)
+    this.generatePath = this.generatePath.bind(this)
+  }
+
+  onSubmit(submitEvent) {
+    submitEvent.preventDefault();
+    const url = this.generatePath(this.state.suggestions[this.state.activeSuggestion])
+    console.log(this.props, this.context);
+    this.clearUserInput()
+    this.context.router.push(url)
+  }
+
+  onChange(inputEvent) {
+    const { suggestion } = this.state
+    const userInput = inputEvent.currentTarget.value
+
+    this.setState({
+      userInput
+    })
+
+    if (userInput.length > 2) {
+      axios({
+        method: 'GET',
+        url: `${appConfig.shepApi}/autosuggest?query=${userInput}`,
+        crossDomain: true,
+        headers: {
+          'Access-Control-Allow-Origin': '*',
+          'Content-Type': 'application/json',
+        },
+      })
+      .then(res => {
+        this.setState({
+          suggestions: res.data.autosuggest
+        })
+      })
+    }
+
+  }
+
+  clearUserInput() {
+    this.setState({
+      userInput: ''
+    })
+  }
+
+  generatePath(item) {
+    const subfield = item.class === 'subfield'
+    const base = appConfig.baseUrl
+    if (subfield) {
+      return `${base}/subject_headings?filter=${item.label}`
+    } else if (item.uuid) {
+      return `${base}/subject_headings/${item.uuid}`
+    }
+  }
+
+  render() {
+    const {
+      onChange,
+      onSubmit,
+      state: {
+        suggestions,
+        activeSuggestion,
+        showSuggestions,
+        userInput
+      }
+    } = this;
+
+    let suggestionsListComponent;
+
+    if (userInput && suggestions.length) {
+      suggestionsListComponent = (
+        <ul className="suggestions">
+        {suggestions.map((suggestion, index) => {
+          return (
+            <AutosuggestItem
+              item={suggestion}
+              generatePath={this.generatePath}
```

Comment:
Am I reading this right that we're passing in `item` and passing `this.generatePath` to `AutosuggestItem`, where it's used to compute `generatePath(item)`? Is there a functional/stylistic reason to not just generate the path for the item before/as it's sent into `AutosuggestItem` (i.e. `<AutosuggestItem ... itemPath={this.generatePath(item)} ...`) rather than execute code in the parent component?